y-lohse/inkjs
# Upgrade TypeScript to v5

## Description

Upgrade TypeScript version on the project (hopefully) without requiring major changes to Rollup and Jest configuration.

The main changes are documented in individual commits (happy to squash these if preferable). Getting it to build correctly required a few minor changes to the source to take into account recent changes to enum behaviour, and dealing with several small issues relating to scope ambiguity.

Some minor warnings with peer dependencies and a bunch of linter stuff in the tests could probably be looked at further.

Need to manually test under the following scenarios:

- Checkout, install and run tests from fresh clone of the repo
- Compile from TS source and build a releasable version of the package
- Install as npm package into separate TS project and successfully import `Ink.Story` and `Ink.Compiler`
- Install as npm package into separate JS project and successfully import `Ink.Story` and `Ink.Compiler`
- Copypasta compiled JS dist files into non NPM project and confirm they work when imported via script tag

## Checklist

- [x] The new code additions passed the tests (`npm test`).
- [ ] The linter ran and found no issues (`npm run-script lint`).
